packages/connect/src/ExtensionProvider/ExtensionProvider.ts

```diff
@@ -96,10 +96,11 @@ export class ExtensionProvider implements ProviderInterface {
     if (parachain) {
       this.#parachainSpecs = parachain
     }
-    this.#commonMessageData = {
-      chainId: nextChainId++,
-      origin: EXTENSION_PROVIDER_ORIGIN,
-    }
+    this.#chainId = nextChainId++
+  }
+
+  public get chainId(): number {
+    return this.#chainId
   }
```

Comment:
Why is this public?

Reply:
honest answer... So that we can test it more easily... I know, a bit of a shame, but the alternative was quite convoluted to make the tests work.

Reply:
> we leave it private

I mean remove the function

Reply:
It's been used in the tests, though. [Here](https://github.com/paritytech/substrate-connect/pull/621/files#diff-162c74379f51c3494d8d9f03d810b9b2a55e688b7b2e72a7e21036516d3c76c3R137) and [here](https://github.com/paritytech/substrate-connect/pull/621/files#diff-162c74379f51c3494d8d9f03d810b9b2a55e688b7b2e72a7e21036516d3c76c3R194).